de.dlr.sc.virsat.graphiti/src/de/dlr/sc/virsat/graphiti/util/DiagramHelper.java

```diff
@@ -88,9 +93,23 @@ public static StructuralElementInstance getOwningStructuralElementInstance(EObje
 			seiUri = seiUri.appendFileExtension(VirSatProjectCommons.FILENAME_EXTENSION);
 			seiUri = URI.createPlatformResourceURI(seiUri.toFileString(), true);
 			
-			Path path = new Path(resourceUri.toPlatformString(false));
-			IResource iResource = ResourcesPlugin.getWorkspace().getRoot().findMember(path);
-			IProject project = iResource.getProject();
+			
+//			Path path = new Path(resourceUri.toPlatformString(false));
+//			IResource iResource = ResourcesPlugin.getWorkspace().getRoot().findMember(path);
+//			IProject project = iResource.getProject();
+//			ResourceSet resourceSet = VirSatResourceSet.getResourceSet(project);
+//			
+			String decodedPath = null;
+			try {
+				decodedPath = URLDecoder.decode(resourceUri.toPlatformString(false), StandardCharsets.UTF_8.name());
+				Activator.getDefault().getLog().info("Decoded path: " + decodedPath);
+			} catch (UnsupportedEncodingException e) {
+				Activator.getDefault().getLog().error("Failed to decode URI", e);
+			}
+			//Path path = new Path(decodedPath);
+			//IResource resource = ResourcesPlugin.getWorkspace().getRoot().findMember(path);
+			IFile file = ResourcesPlugin.getWorkspace().getRoot().getFile(new Path(decodedPath));

```

Comment:
I'm not sure about the control flow here. If the decoding failed, it will be logged, but then the method continues anyway? At this point, `decodedPath` will be `null`. Does it make sense to continue with that or should we rather just `return null` in the `catch` block?